goserver/session/session.go

```diff
@@ -0,0 +1,96 @@
+package session
+
+import (
+	"crypto/hmac"
+	"crypto/sha1"
+	"encoding/base64"
+	"flag"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+)
+
+var sessionCookieSecret = flag.String("session_cookie_secret", "", "secret key used to encrypt the session cookie")
+
+const (
+	sessionLifeInSec = 7200
+	cookieLifeInSec  = 7200
+	cookieName       = "session"
+)
+
+// Session stores necessary information for a telehealth session.
+type Session struct {
+	sid, FHIRURL string
+	expireAt     time.Time
+}
+
+// SessionID the id of the session.
+func (s *Session) SessionID() string {
+	return s.sid
+}
+
+// New creates a new session and set cookie containning the encoded session id.
+func New(m Manager, w http.ResponseWriter, r *http.Request) (*Session, error) {
```

Comment:
It's unclear why some methods are defined at the top level while others are defined as methods on session.Manager. After my refactoring suggestion above, I think it will be simpler to define all of these as methods on session.Manager.

Reply:
See new design